sample-apps/communication/main.js

```diff
@@ -45,3 +50,51 @@ function onNavigateUsingMessageClick () {
   }
   veraApi.sendArmeMessage({ packageName: navigationPackageName, data })
 }
+
+async function onNearestPoiClick() {
+  const MAX_DISTANCE_V = 2
+  const MAX_DISTANCE_H = 3
+  var cameraPose = veraApi.getCameraPose()
+  const filter = {
+    bool: {
+      filter: [{
+        range: {
+          "ar:geometry.center_y": {
+            gte: cameraPose.translation[1] - MAX_DISTANCE_V,
+            lte: cameraPose.translation[1] + MAX_DISTANCE_V
+          }
+        }
+      }]
+    }
+  }
+  // Geo-filter array: camX, camZ, max 2D horizontal distance (m)
+  const geofilter = [cameraPose.translation[0], cameraPose.translation[2], MAX_DISTANCE_H]
+  const allNearbyPois = await veraApi.querySemanticObjects({confKey: appConfig._id, filter, geofilter})
+
+  let nearestPoiName = 'None'
+  if (Object.values(allNearbyPois).length > 0) {
+    // Get 3D distances to camera pos of all result POIs
+    const allPoiDistances = Object.values(allNearbyPois).map(poi => {
+      const camPos = cameraPose.translation
+      const poiPos = poi['ar:geometry']
+      if (!poiPos) return 9999
```

Comment:
`Infinity`, add a comment about this line

Reply:
Done